### DISCUSSION TITLE
关于consul_kv服务发现随机端口的问题

### DISCUSSION THREAD

**Discussion by maybgit** [apache/apisix#8287] [Q&A]
1、APISIX会读取consul_kv的upstream配置使用，假如upstream都是使用动态端口的，每次服务构建后，以前的地址端口都会失效的，那么APISIX会在什么时候清除之前读取到的地址

2、如果A服务当前的地址是 192.168.1.2:12345，B服务的地址是：192.168.1.2:54321，并且这些地址都被APISIX FETCH使用了，那么现在A，B服务将重新构建，A服务的动态端口变成了  192.168.1.2:54321，B服务的动态端口变成了其它的，那么APISIX上一次读取的54321是B服务的，如果APISIX不删旧的配置的话，就会有问题，因为文档说，配置了健康检查，如果健康检查通过，失效的地址会恢复就会引发这个问题。

想咨询下，APISIX是否存在这个问题？因为我们目前的所有服务，都是使用动态端口注册到consul中的。

**Comment by tzssangglass** [comment]
duplicate with #5920?
